faker_nonprofit/__init__.py

```diff
@@ -185,8 +189,30 @@ def nonprofit_name(self):
         suffix = self.random_element(SUFFIXES)
         topic = self.random_element(TOPICS)
 
-        return " ".join([prefix, topic, suffix]).strip()
+        name = " ".join([prefix, topic, suffix]).strip()
+        if suffix == "Foundation" or suffix == "Fund":
+            name = self.foundation_name()
+
+        return name
 
     def nonprofit_title(self):
         """Fake nonprofit job titles."""
         return self.random_element(JOB_TITLES)
+
+    def foundation_name(self):
+        """Fake foundation names."""
+        nameType = self.random_element(['City', 'Other', 'Person', 'State'])
+        prefix = ""
+        if (nameType == 'City'):
+            prefix = self.generator.city()
+        elif (nameType == 'Person'):
+            prefix = self.generator.last_name()
```

Comment:
If we want to add additional variety, this could sometimes just be last_name but sometimes be first_name last_name, or it could sometimes be last_name family (to get things like "Richardson Family Fund" or "Gomez Family Foundation")